src/components/generics/vacation-test-forms/employee-vacation-requests.tsx

```diff
@@ -360,43 +397,102 @@ const RenderEmployeeVacationRequests = ({ persons }: Props) => {
   // };
 
   /**
-   * Handle request type
-   *
-   * @param type Vacation type
+   * Update vacation request status
+   * 
+   * @param selectedStatusObject id of the vacation request status
+   * @param newStatus status to be applied to the updated vacation request status
    */
-  const handleRequestType = (type: VacationType) => {
-    switch (type) {
-      case VacationType.VACATION:
-        return strings.vacationRequests.vacation;
-      case VacationType.PERSONAL_DAYS:
-        return strings.vacationRequests.personalDays;
-      case VacationType.UNPAID_TIME_OFF:
-        return strings.vacationRequests.unpaidTimeOff;
-      case VacationType.MATERNITY_PATERNITY:
-        return strings.vacationRequests.maternityPaternityLeave;
-      case VacationType.SICKNESS:
-        return strings.vacationRequests.sickness;
-      case VacationType.CHILD_SICKNESS:
-        return strings.vacationRequests.childSickness;
-      default:
-        return strings.vacationRequests.vacation;
+  const updateVacationRequestStatus = async (selectedStatusObject: VacationRequestStatus, newStatus: VacationRequestStatuses) => {
+    if (!selectedStatusObject) {
+      return;
+    }
+
+    try {
+      const updateApi = Api.getVacationRequestStatusApi(accessToken?.access_token);
+      const updatedRequestStatus = await updateApi.updateVacationRequestStatus({
+        id: selectedStatusObject.id!,
+        statusId: selectedStatusObject.id!,
```

Comment:
Rather than assert the id's here you could add the check for them to the above condition where selectedStatusObject is already being checked

Reply:
Added check to where selectedStatusObject is already being checked

```suggestion
        statusId: selectedStatusObject.id,
        vacationRequestStatus: {
```